proparse/src/main/java/org/prorefactor/proparse/antlr4/ProgressLexer.java

```diff
@@ -1002,6 +1002,20 @@ public int getCol() {
     public int getSourceNum() {
       return sourceNum;
     }
+
+    @Override
+    public int hashCode() {
+      return (13 * file) + (17 * line) + (31 * col) + (37 * sourceNum);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this.getClass() == obj.getClass()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "obj" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2259)